mlflow/sentence_transformers/__init__.py

```diff
@@ -371,18 +402,58 @@ def predict(self, sentences, params: Optional[Dict[str, Any]] = None):
 
         :return: Model predictions.
         """
-        # When the input is a single string, it is transformed into a DataFrame with one column
-        # and row, but the encode function does not accept DataFrame input
+        # When the input is a single string or a dictionary, it is transformed into a DataFrame with one column
+        # and row. When the input is, but the encode function does not accept DataFrame input
+        convert_output = False
         if type(sentences) == pd.DataFrame:
-            sentences = sentences[0]
+            # Wrap the output to OpenAI format only when the input is dict `{"input": ... }`
+            if list(sentences.columns)[0] == "input":
```

Comment:
Currently the `transformers` flavor only wraps the output when the inference `task` is given, maybe we should also do the same check here, e.g., something like
```suggestion
            if task and list(sentences.columns)[0] == "input":
```